### PR TITLE
use NcReferenceList in whiteboard

### PR DESCRIPTION
**Pull request by grnd-alt**
resolves: #9

### CODE REVIEW

File: src/sidebar/ExampleSidebar.tsx [hunk 1]
```diff
@@ -5,33 +5,38 @@
 
 // https://github.com/excalidraw/excalidraw/blob/4dc4590f247a0a0d9c3f5d39fe09c00c5cef87bf/examples/excalidraw
 
-import { useState } from "react";
```
Comment (juliushaertl) [inline src/sidebar/ExampleSidebar.tsx:8]: Please make sure to rebase to latest main branch, this file should no longer exist ;)

**Comment by juliushaertl** [comment]
Forgot to paste from our call notes:

- [ ] Test different file types and styling
- [ ] Check if height can be made fit the full frame for interactive widgets
- [ ] Check react hooks loop issue

File: src/collaboration/util.ts [hunk 2]
```diff
@@ -38,6 +68,13 @@ export function shouldDiscardRemoteElement(localAppState: Readonly<AppState>, lo
 	) {
 		return true
 	}
+
+	// embeddables get updated a lot by the excalidraw library when collaboration is active
+	// we need to filter out useless updates to not rerender every millisecond.
+	// current master of the excalidraw library probably fixes this issue but it's not available in excalidraw@latest (0.17.6)
```
Comment (grnd-alt) [inline src/collaboration/util.ts:74]: checking if the objects are DeepEqual is a workaround. Not entirely certain, but I do think if we could use the lib like excalidraw.com does here: https://github.com/excalidraw/excalidraw/blob/master/excalidraw-app/collab/Collab.tsx#L751 we would not need this workaround. However I couldn't figure out how to use the master version of excalidraws library in our build process.

Comment (juliushaertl) [reply src/collaboration/util.ts:74]: Sounds good, being able to build from master or another branch/a fork there is something that could certainly be useful, but we can check that separately

**Review by juliushaertl — APPROVED** [review]
Tested and works like a charm 👍

**Pull request merged**